pkgs/scorec.yaml

```diff
@@ -20,30 +20,73 @@ build_stages:
+      -DSCOREC_CXX_WARNINGS:BOOL=OFF
+      -DENABLE_ZOLTAN:BOOL=ON
+      -DSCOREC_CXX_OPTIMIZE:BOOL=ON
+      -DMDS_ID_TYPE:STRING=int
+      -DPCU_COMPRESS=ON
+      -DENABLE_FPP=ON
+      -DENABLE_SIMMETRIX:BOOL=OFF
+      -DBZIP2_INCLUDE_DIR:PATH=${BZIP2_DIR}/include
+      -DBZIP2_LIBRARY_RELEASE:PATH=${BZIP2_DIR}/lib/libbz2.so
+      -DZOLTAN_PREFIX:PATH=${ZOLTAN_DIR}
+      -DPARMETIS_PREFIX:PATH=${PARMETIS_DIR}
+      -DCMAKE_BUILD_TYPE:STRING=Release"
+
+- when: machine == 'CrayXE6' or machine == 'CrayXC30' or machine == 'CrayXC40'
+  name: cmake-flags-cray
+  after: cmake-flags
+  before: configure
+  handler: bash
+  bash: |
+    CMAKE_FLAGS="
+      -DBUILD_SHARED_LIBS:BOOL=OFF
+      -DBUILD_EXES:BOOL=ON
+      -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON
+      -DCMAKE_C_COMPILER:STRING=${MPICC}
+      -DCMAKE_CXX_COMPILER:STRING=${MPICXX}
+      -DCMAKE_EXE_LINKER_FLAGS_RELEASE:STRING=-dynamic
+      -DCMAKE_SHARED_LINKER_FLAGS_RELEASE:STRING=-dynamic
+      -DCMAKE_MODULE_LINKER_FLAGS_RELEASE:STRING=-dynamic
+      -DSCOREC_CXX_WARNINGS:BOOL=OFF
+      -DENABLE_ZOLTAN:BOOL=ON
+      -DSCOREC_CXX_OPTIMIZE:BOOL=ON
+      -DMDS_ID_TYPE:STRING=int
+      -DPCU_COMPRESS=ON
+      -DENABLE_FPP=ON
+      -DENABLE_SIMMETRIX:BOOL=OFF
+      -DBZIP2_INCLUDE_DIR:PATH=/usr/include
+      -DBZIP2_LIBRARY_RELEASE:PATH=/usr/lib64/libbz2.so
+      -DZOLTAN_PREFIX:PATH=${ZOLTAN_DIR}
+      -DPARMETIS_INCLUDE_DIR:PATH=${PARMETIS_DIR}/include
+      -DPARMETIS_LIBRARY_RELEASE:PATH=${PARMETIS_DIR}/lib/libparmetis.so;${PARMETIS_DIR}/lib/libmetis.so
+      -DCMAKE_BUILD_TYPE:STRING=Release"
+
 - name: configure
-  debug: true
-  extra: [
-  '-DBUILD_SHARED_LIBS:BOOL=OFF',
-  '-DBUILD_EXES:BOOL=ON',
-  '-DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON',
-  '-DCMAKE_C_COMPILER:STRING=${MPICC}',
-  '-DCMAKE_CXX_COMPILER:STRING=${MPICXX}',
-  '-DCMAKE_EXE_LINKER_FLAGS_RELEASE:STRING=-dynamic',
-  '-DCMAKE_SHARED_LINKER_FLAGS_RELEASE:STRING=-dynamic',
-  '-DCMAKE_MODULE_LINKER_FLAGS_RELEASE:STRING=-dynamic',
-  '-DSCOREC_CXX_WARNINGS:BOOL=OFF',
-  '-DENABLE_ZOLTAN:BOOL=ON',
-  '-DSCOREC_CXX_OPTIMIZE:BOOL=ON',
-  '-DMDS_ID_TYPE:STRING="int"',
-  '-DPCU_COMPRESS=ON',
-  '-DENABLE_FPP=ON',
-  '-DENABLE_SIMMETRIX:BOOL=OFF',
-#  '-DBZIP2_PREFIX:PATH=${BZIP2_DIR}',
-  '-DBZIP2_INCLUDE_DIR:PATH=/usr/include',
-  '-DBZIP2_LIBRARY_RELEASE:PATH=/usr/lib64/libbz2.so',
-  '-DZOLTAN_PREFIX:PATH=${ZOLTAN_DIR}',
-#  '-DPARMETIS_PREFIX:PATH=${PARMETIS_DIR}',
-  '-DPARMETIS_INCLUDE_DIR:PATH=${PARMETIS_DIR}/include',
-  '-DPARMETIS_LIBRARY_RELEASE:PATH="${PARMETIS_DIR}/lib/libparmetis.so;${PARMETIS_DIR}/lib/libmetis.so"',
-  '-DCMAKE_BUILD_TYPE:STRING="Release"',
-  ]
+  after: cmake-flags
+  handler: bash
+  bash: |
+    ${CMAKE} ${BUILD} ${CMAKE_FLAGS}
+
+- name: make
+  after: configure
+  handler: bash
+  bash: |
+    ${CMAKE} --build . -- -j ${HASHDIST_CPU_COUNT}
+
+- name: install
+  after: make
+  handler: bash
+  bash: |
+    ${CMAKE} --install .
+
```

Comment:
is this section made necessary because of the `when` statements?

If I remember correctly, the scorec executables usually have a hard time building without the shared libraries - not sure if you tried calling an executable on onyx, like `zsplit`. Otherwise, I think making platform specific configurations is a good idea for the `scorec` package.

Reply:
Yes. It's possible that we could get it working without overriding those, but I went ahead and exposed the definitions because I had to change a few things.

I think I have not been using the scorec executable tools on onyx. I suspect we can get them going, but I'll leave this alone for now.